src/engine/math_util.c

```diff
@@ -541,47 +617,76 @@ void mtxf_lookat(Mat4 mtx, Vec3f from, Vec3f to, s32 roll) {
  * 'mtx' is the look-at matrix from the camera.
  * 'position' is the position of the object in the world.
  * 'scale' is the scale of the object.
- * 'angle' rotates the object while still facing the camera.
+ * 'roll' rotates the object while still facing the camera.
  */
-void mtxf_billboard(Mat4 dest, Mat4 mtx, Vec3f position, Vec3f scale, s32 angle) {
-    register s32 i;
-    register f32 sx = scale[0];
-    register f32 sy = scale[1];
-    register f32 sz = ((f32 *) scale)[2];
-    register f32 *temp2, *temp = (f32 *)dest;
-    for (i = 0; i < 16; i++) {
-        *temp = 0;
-        temp++;
+void mtxf_billboard(Mat4 dest, Mat4 src, Vec3f position, Vec3f scale, s32 roll) {
+    f32 sx = scale[0];
+    f32 sy = scale[1];
+    f32 sz = scale[2];
+    f32 *dstp = (f32 *)dest;
+    s32 i;
+    for (i = 0; i < 12; i++) {
+        *dstp++ = 0;
     }
-    if (angle == 0x0) {
+    if (roll == 0x0) {
         // ((u32 *) dest)[0] = FLOAT_ONE;
-        dest[0][0] = sx; // [0][0]
+        dest[0][0] = sx;
         dest[0][1] = 0;
         dest[1][0] = 0;
         // ((u32 *) dest)[5] = FLOAT_ONE;
-        dest[1][1] = sy; // [1][1]
+        dest[1][1] = sy;
     } else {
-        dest[0][0] = (coss(angle) * sx);
-        dest[0][1] = (sins(angle) * sx);
-        dest[1][0] = (-dest[0][1] * sy);
-        dest[1][1] = ( dest[0][0] * sy);
+        dest[0][0] =  coss(roll) * sx;
+        dest[0][1] =  sins(roll) * sx;
+        dest[1][0] = -dest[0][1] * sy;
+        dest[1][1] =  dest[0][0] * sy;
     }
     // ((u32 *) dest)[10] = FLOAT_ONE;
-    // dest[2][2] = sz; // [2][2]
-    ((f32 *) dest)[10] = sz; // [2][2]
-    dest[2][3] = 0;
-    ((u32 *) dest)[15] = FLOAT_ONE; // [3][3]
+    dest[2][2] = sz;
+    linear_mtxf_mul_vec3f_and_translate(src, dest[3], position);
+    ((u32 *) dest)[15] = FLOAT_ONE;
+}
```

Comment:
this is a bit of a mess here in a widely used function, theres several lines commented out and you also added a very major function call here (`linear_mtxf_mul_vec3f_and_translate`) so its pretty concerning that this functions half baked

Reply:
OOH so the diff got all jacked up since there aren't any newlines, here's my tip:
in vscode go to compare changes mode, and try to make functions line up. sounds a little weird but you could probably add a newline above and below the call for `linear_mtxf_mul_vec3f_and_translate` and it would make the rest of the diff line up